lib/array_intersection.rb

```diff
@@ -1,3 +1,10 @@
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
-end
\ No newline at end of file
+  common_items = []
+  list1.each do |item|
+    if common_items.include?(item) == false && list2.include?(item) == true
```

Comment:
`.include?` is an O(n) method and since you have this in a loop `intersection` becomes an O(n * m) method.